tools/psa/mbed_spm_tfm_common.py

```diff
@@ -599,31 +593,37 @@ def validate_partition_manifests(manifests):
 
 
 def is_test_manifest(manifest):
-    return 'tests' in manifest
+    return 'TESTS' in manifest
 
 
 def is_service_manifest(manifest):
     return not is_test_manifest(manifest)
 
 
-def manifests_discovery(root_dir=SERVICES_DIR):
+def manifests_discovery(root_dirs, ignore_paths):
     service_manifest_files = set()
     test_manifest_files = set()
+    for root_dir in root_dirs:
+        for root, dirs, files in os.walk(root_dir, followlinks=True):
+            # Filters paths if they are inside one of the ignore paths
+            if next((True for igp in ignore_paths if igp in root), False):
+                continue
 
-    for root, dirs, files in os.walk(root_dir):
-        to_add = [path_join(root, f) for f in
-                  fnmatch.filter(files, MANIFEST_FILE_PATTERN)]
-        service_manifest_files.update(filter(is_service_manifest, to_add))
-        test_manifest_files.update(filter(is_test_manifest, to_add))
+            to_add = [path_join(root, f) for f in
+                      fnmatch.filter(files, MANIFEST_FILE_PATTERN)]
+            service_manifest_files.update(filter(is_service_manifest, to_add))
+            test_manifest_files.update(filter(is_test_manifest, to_add))
 
-    return sorted(list(service_manifest_files)), sorted(list(test_manifest_files))
+    service_manifest_files = sorted(list(service_manifest_files))
+    test_manifest_files = sorted(list(test_manifest_files))
+    return service_manifest_files, test_manifest_files
```

Comment:
this PR adds the option for the user to add his own secure partitions
that why we need to scan the entire tree

Reply:
I understand that. But considering we already scan the entire tree in the build system (you can get at these files through the Resource object that's part of the toolchain), I was wondering if you could make use of that scan instead of scanning again? Perhaps its not possible due to dependencies or something, but I just wanted to ask.

Reply:
the resources object doesn't work for us when we need to discover partitions in tests
when we are building tests we need to discover all the partitions before building the OS, the resources object wont find them

Reply:
There are multiple resource scans during build process. Especially during build test flow. There is one full tree scan just for determine baud rate.
None of the conventional scans suite our requirements as we need to discover ALL the manifest files from ALL test folders and application at once for code generation and manifest validation.

Reply:
@orenc17 @alzix good point, the test build scan is not particularly efficient. In that case lets give this a go then.

> There is one full tree scan just for determine baud rate.

Yikes, that doesn't sound right/intentional. I'd be interested in hearing more details about that.